worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/TaskStowingWorker.java

```diff
@@ -0,0 +1,133 @@
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskStowingWorker.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private final DatabaseClient databaseClient;
+
+    public TaskStowingWorker(final DatabaseClient databaseClient)
+    {
+        this.databaseClient = databaseClient;
+    }
+
+    @Override
+    public void checkHealth(final HealthMonitor healthMonitor)
+    {
+        try {
+            databaseClient.checkHealth();
+        } catch (final Exception exception) {
+            final String message = "Database is unhealthy";
+            LOGGER.error(message, exception);
+            healthMonitor.reportUnhealthy(message);
+        }
+    }
+
+    @Override
+    public void processDocument(final Document document) throws InterruptedException, DocumentWorkerTransientException
+    {
+        // TODO: Check forwarding behavior in WorkerCore, i.e. isTaskIndendedForThisWorker - this worker currently won't get this task.
+
+        final WorkerTaskData workerTaskData = document.getTask().getService(WorkerTaskData.class);
+        if (workerTaskData == null) {
+            processFailure(TaskStowingWorkerFailure.WORKER_TASK_DATA_NOT_PRESENT, document);
+            return;
+        }
+
+        final TrackingInfo trackingInfo = workerTaskData.getTrackingInfo();
+        if (trackingInfo == null) {
+            processFailure(TaskStowingWorkerFailure.TRACKING_INFO_NOT_PRESENT, document);
+            return;
+        }
+
+        final String jobTaskIdFromTrackingInfo = trackingInfo.getJobTaskId();
+        if (Strings.isNullOrEmpty(jobTaskIdFromTrackingInfo)) {
+            processFailure(TaskStowingWorkerFailure.JOB_TASK_ID_NOT_PRESENT, document);
+            return;
+        }
+
+        final JobTaskId jobTaskId = JobTaskId.fromMessageId(jobTaskIdFromTrackingInfo);
+
+        final String partitionId = jobTaskId.getPartitionId();
+        if (Strings.isNullOrEmpty(partitionId)) {
+            processFailure(TaskStowingWorkerFailure.FAILED_TO_PARSE_PARTITION_ID_FROM_JOB_TASK_ID, document);
+            return;
+        }
+
+        final String jobId = jobTaskId.getJobId();
+        if (Strings.isNullOrEmpty(jobId)) {
+            processFailure(TaskStowingWorkerFailure.FAILED_TO_PARSE_JOB_ID_FROM_JOB_TASK_ID, document);
+            return;
+        }
+
+        try {
+            databaseClient.stowTask(
+                partitionId,
+                jobId,
+                workerTaskData.getClassifier(),
+                workerTaskData.getVersion(),
+                workerTaskData.getData(),
+                workerTaskData.getStatus().name(),
+                workerTaskData.getContext() != null ? workerTaskData.getContext()
+                : OBJECT_MAPPER.writeValueAsBytes(Collections.<String, byte[]>emptyMap()), // TODO check this: workerTaskData.getContext() always seems to be null, even when the task sent to the worker includes a non-null context?
```

Comment:
I saw some unexpected (I think) behaviour when passing a non-null `context` value to the worker in a task message. 

It seems that even when the `context` contained in the task message sent to Rabbit was non-null, by the time the worker receives the document here, `workerTaskData.getContext()` will always be null - it seems the `context` value is being discarded at some point. 

I had a quick look through the worker-framework and saw this `inludeTaskContext` flag, so maybe it's something to do with that?

https://github.com/WorkerFramework/worker-framework/blob/570c5774f30bb488a12781f6d9d53e593ce99feb/worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java#L152

```
    @Override
    public void addResponse(final WorkerResponse response, final boolean includeTaskContext)
    {
        Objects.requireNonNull(response);
        Objects.requireNonNull(response.getQueueReference());
        //increment the task an sub task response count
        incrementResponseCount(false);
        
        final TaskMessage responseMessage = createResponseMessage(includeTaskContext, response);

        singleMessageBuffer.add(responseMessage);
    }
```

I have modelled `context` to be non null in the database, as it is annotated as that inside the `TaskMessage` object:

https://github.com/WorkerFramework/worker-framework/blob/develop/worker-api/src/main/java/com/hpe/caf/api/worker/TaskMessage.java

```
@NotNull
private Map<String, byte[]> context;
```

so having to do some null to empty translation here before saving it in the database.

Reply:
I think this is ok now, I think I was setting the context key to an invalid value in my test:

Looking at the Worker Framework, when it executes a task, the [WorkerExecutor.java](https://github.com/WorkerFramework/worker-framework/blob/570c5774f30bb488a12781f6d9d53e593ce99feb/worker-core/src/main/java/com/hpe/caf/worker/core/WorkerExecutor.java#L75) uses the task message to create a [WorkerTaskImpl.java](https://github.com/WorkerFramework/worker-framework/blob/570c5774f30bb488a12781f6d9d53e593ce99feb/worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java#L45]):

```
final WorkerTaskImpl workerTask = createWorkerTask(taskInformation, tm, poison, headers, codec);
```

and the `getContext` method is implemented as:

```
@Override
public byte[] getContext()
{
   return taskMessage.getContext().get(servicePath.toString());
}
```

So, the context needs to have the `caf/worker` key in order to be passed along.